### PR TITLE
Document application_name parameter

### PR DESCRIPTION
**Pull request by eulerto**
It also fixes a typo.

### CODE REVIEW

File: doc/config.rst [hunk 1]
```diff
@@ -394,7 +394,7 @@ stats_users
 -----------
 
 Comma-separated list of database users that are allowed to connect and
-run read-only queries on console. Thats means all SHOW commands except
+run read-only queries on console. That means all SHOW commands except
```
Comment (petere) [inline doc/config.rst]: I've committed that separately.

File: doc/config.rst [hunk 2]
```diff
@@ -897,6 +897,13 @@ max_db_connections
 Configure a database-wide maximum (i.e. all pools within the database will
 not have more than this many server connections).
 
+application_name
+----------------
+
+Set application name on connection start.  This logic applies only on start of
+connection, if application_name is later changed with SET, pgbouncer does not
```
Comment (petere) [inline doc/config.rst]: What do you mean by "changed with SET"? Do you mean if it's changed later by reloading the config file?

Comment (eulerto) [reply doc/config.rst]: Ooops. I mean "changed by SET".

Comment (petere) [reply doc/config.rst]: Still don't get it. `SET` is a PostgreSQL command, not pgbouncer. What is the scenario you are describing here?

Comment (eulerto) [reply doc/config.rst]: @petere I use the same sentence of `application_name_add_host` parameter in config.rst. If you don't get it, let's try to rephrase both such as:

> pgbouncer sets application_name when it opens a connection. If a client changes application_name via SET command then pgbouncer will not reset it.

Comment (curlup) [reply doc/config.rst]: @petere, I believe it's about PgBouncer's `SET command` 

from docs:
```
SET key = arg;
Changes a configuration setting (see also SHOW CONFIG). For example:

SET log_connections = 1;
```
here 
https://github.com/pgbouncer/pgbouncer/blob/master/doc/usage.rst#set-key--arg
or here
https://pgbouncer.github.io/usage.html#other-commands

**Pull request closed without merging**